egui_demo_lib/src/apps/demo/zoom_rotate.rs

```diff
@@ -0,0 +1,117 @@
+        Frame::dark_canvas(ui.style()).show(ui, |ui| {
+            // Note that we use `Sense::drag()` although we do not use any pointer events.  With
+            // the current implementation, the fact that a touch event of two or more fingers is
+            // recognized, does not mean that the pointer events are suppressed, which are always
+            // generated for the first finger.  Therefore, if we do not explicitly consume pointer
+            // events, the window will move around, not only when dragged with a single finger, but
+            // also when a two-finger touch is active.  I guess this problem can only be cleanly
+            // solved when the synthetic pointer events are created by egui, and not by the
+            // backend.
+
+            // set up the drawing canvas with normalized coordinates:
+            let (response, painter) =
+                ui.allocate_painter(ui.available_size_before_wrap_finite(), Sense::drag());
+            // normalize painter coordinates to ±1 units in each direction with [0,0] in the center:
+            let painter_proportions = response.rect.square_proportions();
+            let to_screen = RectTransform::from_to(
+                Rect::from_min_size(Pos2::ZERO - painter_proportions, 2. * painter_proportions),
+                response.rect,
+            );
+
+            // check for touch input (or the lack thereof) and update zoom and scale factors, plus
+            // color and width:
+            let mut stroke_width = 1.;
+            let mut color = Color32::GRAY;
+            if let Some(multi_touch) = ui.input().multi_touch() {
+                // This adjusts the current zoom factor and rotation angle according to the dynamic
+                // change (for the current frame) of the touch gesture:
+                self.zoom *= multi_touch.zoom_delta;
+                self.rotation += multi_touch.rotation_delta;
+                // touch pressure shall make the arrow thicker (not all touch devices support this):
+                stroke_width += 10. * multi_touch.force;
+                // the drawing color depends on the number of touches:
+                color = match multi_touch.num_touches {
+                    2 => Color32::GREEN,
+                    3 => Color32::BLUE,
+                    4 => Color32::YELLOW,
+                    _ => Color32::RED,
+                };
+                // for a smooth touch experience (not strictly required, but I had the impression
+                // that it helps to reduce some lag, especially for the initial touch):
+                ui.ctx().request_repaint();
+            } else if let Some(last_time) = self.time_of_last_update {
+                // This has nothing to do with the touch gesture. It just smoothly brings the
+                // painted arrow back into its original position, for a nice visual effect:
+                let dt = ui.input().time - last_time;
```

Comment:
You can use `ui.input().unstable_dt` instead here (not the best name, I know) so you don't need `time_of_last_update`

```suggestion
                let dt = ui.input().unstable_dt;
```

Reply:
resolved by https://github.com/emilk/egui/pull/306/commits/e3cc56ef6f6f6297312c3ba056d217ef5bea81a9